### PR TITLE
Harden blueprint updates with revision safety, deterministic fetch policy, reload recovery and compatibility testing

### PR DESCRIPTION
**Pull request by luuquangvu**
## Summary by Sourcery

Strengthen blueprint update safety and determinism, ensuring filesystem revisions, background refresh generation, and reload flows are handled robustly while tightening network/encoding policy and selector compatibility checks.

New Features:
- Track pending domain reloads across restarts and automatically retry failed reloads until they succeed.
- Enforce per-blueprint ownership tokens for background refresh work to prevent obsolete generations from mutating current state.
- Capture and enforce filesystem revision preconditions for imports, installs, and restores so external edits or races cannot clobber user changes.

Bug Fixes:
- Ensure metadata only persists for real, undeleted blueprint files and skips symlink aliases to avoid duplicate or unsafe paths.
- Fix connection timeout handling in the safe network backend by sharing remaining budget fairly across resolved addresses and falling back after timeouts.
- Decode provider responses as strict UTF-8 with BOM handling and surface invalid bytes as explicit errors instead of silently replacement-decoding.

Enhancements:
- Extend blueprint metadata to track a raw local file hash separately from the semantic hash to support revision checks and auto-update safety.
- Make background refresh generation-aware, cancelling obsolete tasks and broadcasting coordinator updates once per completed batch.
- Refine selector mismatch detection by comparing normalized selector configurations so domain changes in entity selectors are treated as compatibility risks.
- Improve error reporting and retry behavior around remote fetch policy violations using a dedicated BlueprintFetchPolicyError and sanitized log messages.
- Normalize blueprint input selector configs to a stable, presentation-free structure for more deterministic risk analysis.

Documentation:
- Clarify and modernize English and Vietnamese README sections around installation, setup, backup/restore semantics, requirements, and licensing wording.

Tests:
- Add comprehensive filesystem revision safety tests covering external edits, concurrent creation/deletion, symlink targets, and restore preconditions.
- Expand coordinator tests for symlink scanning, background refresh generation semantics, pending reload recovery, 304 auto-update with raw hashes, and selector constraint changes.
- Update networking tests to cover shared timeout budgeting, connect-timeout fallback, strict size limits, and HTTPS enforcement via the new fetch policy error type.

## Summary by CodeRabbit

* **New Features**
  * Generation-aware background refresh with persisted “pending reload” domain tracking and improved reload reconciliation.
  * Revision-aware blueprint install/restore using preconditions to prevent conflicting local edits.

* **Bug Fixes**
  * Safer blueprint file updates/restores with stronger mismatch detection, including rejection when the final path is a symlink.
  * Hardened blueprint fetching (redirect/HTTPS/size checks), stricter UTF-8 handling, and redacted source URL logging.
  * Improved selector compatibility/risk detection via normalized selector configuration.

* **Documentation**
  * Standardized README headings/ToC anchors and added section separators (including translated docs).

* **Tests**
  * Expanded coverage for refresh generations, revision mismatch handling, persistence across restarts, networking/decoding, and selector edge cases.

### CODE REVIEW

**Comment by luuquangvu** [comment]
@coderabbitai review

**Comment by luuquangvu** [comment]
@sourcery-ai review

**Comment by luuquangvu** [comment]
@coderabbitai review

**Comment by luuquangvu** [comment]
@coderabbitai review

File: tools/validate_compatibility.py [hunk 1]
```diff
@@ -524,23 +547,74 @@ def _format_cmd_str(cmd: Any) -> str:
...
+        try:
+            published_versions.append((Version(raw_version), raw_version))
+        except InvalidVersion:
+            continue
+    if not published_versions:
+        raise ValueError(f"PyPI returned no published {package_name} releases")
+    return _validate_version_label(
+        f"{normalize_package_name(package_name)}_version",
+        max(published_versions)[1],
+    )
```
[the remark this answers is not in the document]
Comment (luuquangvu) [reply tools/validate_compatibility.py:570]: This design intentionally includes the beta version.
